index.ts

```diff
@@ -642,7 +642,7 @@ export class HttpClient {
         maxSockets: maxSockets,
         keepAlive: this._keepAlive,
         proxy: {
-          proxyAuth: `${proxyUrl.username}:${proxyUrl.password}`,
+          ...((proxyUrl.username || proxyUrl.password) && { proxyAuth: `${proxyUrl.username}:${proxyUrl.password}` }),
```

Comment:
the change looks good. On a related note, I wonder if it should be proxyUrl.username && proxyUrl.password?  if username is set but password isn't (or vice versa) would we want to set it?  The other option is to just check if proxyUrl.username is set because apparently a nul password is valid (see [mozilla bug](https://bugzilla.mozilla.org/show_bug.cgi?id=32761))

Reply:
I would suggest setting it if either username or password is defined. It's up to the given authentication backend to handle the nuances. For example, authenticating with Github via only a token (in the password field) is perfectly valid.

Reply:
makes sense.